tfupdate provider returns panic

I am getting the following error:

```shell
$  docker run -it --rm -v $PWD:/work --entrypoint /bin/bash minamijoyo/tfupdate

bash-5.0# tfupdate provider -v 3.63.0 -r aws /work
panic: runtime error: index out of range [19] with length 19

goroutine 1 [running]:
github.com/minamijoyo/tfupdate/tfupdate.(*ProviderUpdater).updateTerraformRequiredProvidersBlockAsObject(0xc00007eae0, 0xc00009bdc0, 0xc0000d3860, 0xc0001b8c70, 0xa2dac0)
        /work/tfupdate/provider.go:120 +0x27f
github.com/minamijoyo/tfupdate/tfupdate.(*ProviderUpdater).updateTerraformBlock(0xc00007eae0, 0xc000228540, 0x7ffcb818feef, 0xc)
        /work/tfupdate/provider.go:72 +0x1f3
github.com/minamijoyo/tfupdate/tfupdate.(*ProviderUpdater).Update(0xc00007eae0, 0xc000228540, 0x7ffcb818feeb, 0x3)
        /work/tfupdate/provider.go:38 +0x39
github.com/minamijoyo/tfupdate/tfupdate.UpdateHCL(0xbd78c0, 0xc000010700, 0xbd6de0, 0xc0001c3bc0, 0x7ffcb818feef, 0xc, 0xb1cd06, 0x8, 0x7ffcb818feeb, 0x3, ...)
        /work/tfupdate/update.go:58 +0x1dd
github.com/minamijoyo/tfupdate/tfupdate.UpdateFile(0xbe6d78, 0xf459f0, 0x7ffcb818feef, 0xc, 0xb1cd06, 0x8, 0x7ffcb818feeb, 0x3, 0x7ffcb818fee1, 0x6, ...)
        /work/tfupdate/file.go:26 +0x298
github.com/minamijoyo/tfupdate/tfupdate.UpdateFileOrDir(0xbe6d78, 0xf459f0, 0x7ffcb818feef, 0xc, 0xb1cd06, 0x8, 0x7ffcb818feeb, 0x3, 0x7ffcb818fee1, 0x6, ...)
        /work/tfupdate/file.go:114 +0x1fa
github.com/minamijoyo/tfupdate/command.(*ProviderCommand).Run(0xc0001afdc0, 0xc000020090, 0x5, 0x5, 0xc0001c37a0)
        /work/command/provider.go:68 +0x579
github.com/mitchellh/cli.(*CLI).Run(0xc0000d4c80, 0xc0000d4c80, 0xc00019ff68, 0x1)
        /go/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x41a
main.main()
        /work/main.go:48 +0x1d0
 


```

--- a/tfupdate/provider.go
+++ b/tfupdate/provider.go
@@ -117,7 +117,13 @@ func (u *ProviderUpdater) updateTerraformRequiredProvidersBlockAsObject(p *hclwr
 
 	i := 0
 	// find key of version
-	for !(tokens[i].Type == hclsyntax.TokenIdent && string(tokens[i].Bytes) == "version") {
+	// Although not explicitly stated in the required_providers documentation,
+	// a TokenQuotedLit is also valid token. Strict speaking there are more
+	// variants because the left hand side of object key accepts an expression in
+	// HCL. For accurate implementation, it should be implemented using the
+	// original parser.
+	for !((tokens[i].Type == hclsyntax.TokenIdent || tokens[i].Type == hclsyntax.TokenQuotedLit) &&
+		string(tokens[i].Bytes) == "version") {
 		i++
 	}
 

--- a/tfupdate/provider_test.go
+++ b/tfupdate/provider_test.go
@@ -323,6 +323,31 @@ terraform {
 terraform {
   required_providers {
     aws = {
+      "version" = "2.65.0"
+      "source"  = "hashicorp/aws"
+    }
+  }
+}
+`,
+			name:    "aws",
+			version: "2.66.0",
+			want: `
+terraform {
+  required_providers {
+    aws = {
+      "version" = "2.66.0"
+      "source"  = "hashicorp/aws"
+    }
+  }
+}
+`,
+			ok: true,
+		},
+		{
+			src: `
+terraform {
+  required_providers {
+    aws = {
       version = "2.65.0"
       source  = "hashicorp/aws"
 

